api/v1alpha1/cloudstorage_types.go

```diff
@@ -42,8 +42,11 @@ type CloudStorageSpec struct {
 	// region for the bucket to be in, will be us-east-1 if not set.
 	Region string `json:"region,omitempty"`
 	// provider is the provider of the cloud storage
-	// +kubebuilder:validation:Enum=aws
+	// +kubebuilder:validation:Enum=aws;azure;gcp
 	Provider CloudStorageProvider `json:"provider"`
+	// config is provider-specific configuration options
+	// +kubebuilder:validation:Optional
+	Config map[string]string `json:"config,omitempty"`
```

Comment:
Should we add validation tags for any required fields per provider ?

Reply:
it's optional.. no required field.

Reply:
Yes, what I meant was, should we make some of the config options mandatory?